src/python/analyzer_executor/src/analyzer_executor_lib/analyzer_executor.py

```diff
@@ -171,26 +181,41 @@ def check_caches(
 
             return False
 
+    def to_event_hash(self, components: Iterable[str]) -> str:
+        joined = ",".join(components)
+        event_hash = hashlib.sha256(joined.encode()).hexdigest()
+        return event_hash
+
     def check_msg_cache(self, file: str, node_key: str, msg_id: str) -> bool:
-        to_hash = str(file) + str(node_key) + str(msg_id)
-        event_hash = hashlib.sha256(to_hash.encode()).hexdigest()
+        event_hash = self.to_event_hash((file, node_key, msg_id))
         return bool(self.message_cache.get(event_hash))
 
     def update_msg_cache(self, file: str, node_key: str, msg_id: str) -> None:
-        to_hash = str(file) + str(node_key) + str(msg_id)
-        event_hash = hashlib.sha256(to_hash.encode()).hexdigest()
+        event_hash = self.to_event_hash((file, node_key, msg_id))
         self.message_cache.set(event_hash, "1")
 
+    def delete_msg_cache(self, file: str, node_key: str, msg_id: str) -> None:
+        """
```

Comment:
if this is the docstring, let's just make it test utility code

Reply:
Done